### PR TITLE
Add omnibus-nikos build image

### PR DESCRIPTION
**Pull request by ISauve**
Creates a new build image which can be used to build the [omnibus-nikos project](https://github.com/DataDog/omnibus-nikos) on a system with an older version of glibc.

Note: an ECR repository was created for this image here: https://github.com/DataDog/cloudops/pull/24122

### CODE REVIEW

File: omnibus-nikos/automake.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+diff -bur automake-1.16/bin/automake.in  automake-1.16.new/bin/automake.in
```
Comment (mx-psi) [inline omnibus-nikos/automake.patch:1]: Can you add a comment above this line with the source of the patch and what it does?

File: omnibus-nikos_x64/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+RUN conda install -c psi4 gcc-5
+
+# Python 3.8
+RUN curl -O -k https://www.python.org/ftp/python/3.8.0/Python-3.8.0.tar.xz && \
+        tar -xf Python-3.8.0.tar.xz -C /tmp/ && \
+        cd /tmp/Python-3.8.0 && \
+        ./configure && \
+        make -j 8 && \
+        make install
```
Comment (mx-psi) [inline omnibus-nikos_x64/Dockerfile:71]: Why do we need to install conda and then install Python from source?

File: omnibus-nikos_x64/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+RUN python -m pip install awscli
+RUN python -m pip install meson
+RUN python -m pip install ninja
```
Comment (mx-psi) [inline omnibus-nikos_x64/Dockerfile:75]: nit (to reduce number of layers)
```suggestion
RUN python -m pip install awscli meson ninja
```
Also I would pin the version that you use

File: omnibus-nikos-automake.patch [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Automake 1.16 attempts to import the 'none' function from perl's List::Util module;
```
Comment (mx-psi) [inline omnibus-nikos-automake.patch:1]: If this is just for this image, then move it to the omnibus-nikos_x64 folder

File: omnibus-nikos_x64/Dockerfile (hunk 2 above)
Comment (ISauve) [reply omnibus-nikos_x64/Dockerfile]: Good point, this is leftover from before I added the conda installation, I'll take it out & just use conda to install python 3.8

**Pull request merged**